internal/app/upstream/client_test.go

```diff
@@ -358,6 +362,62 @@ func TestOpenStreamShouldRetryIfSendFails(t *testing.T) {
 	})
 }
 
+func TestStreamShouldRetryWhenTimeoutMet(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	responseChan := make(chan *v2.DiscoveryResponse)
+	scope := stats.NewMockScope("mock")
+	client := createMockClientWithResponse(ctx, time.Second, responseChan, func(m interface{}) error {
+		return nil
+	}, scope, 1)
+
+	_, done := client.OpenStream(
+		transport.NewRequestV2(&v2.DiscoveryRequest{
+			TypeUrl: resource.ListenerType,
+			Node:    &core.Node{},
+		}), "aggregated_key")
+	defer done()
+	for {
+		var val int64 = 0
+		if scope.Snapshot().Counters() != nil &&
+			scope.Snapshot().Counters()["mock.lds.stream_retry+key=aggregated_key"] != nil {
+			val = scope.Snapshot().Counters()["mock.lds.stream_retry+key=aggregated_key"].Value()
+		}
+		if val > 0 {
+			break
+		}
+	}
+	done()
```

Comment:
One question for my understanding: why do we call `defer done()` if we run `done()` here?

Reply:
Good catch I think the second done is not necessary.